legacy/src/main/kotlin/at/asitplus/gradle/CommonDependencies.kt

```diff
@@ -2,60 +2,125 @@
 
 package at.asitplus.gradle
 
-import AspVersions
+import org.gradle.api.Project
 import org.jetbrains.kotlin.gradle.plugin.KotlinDependencyHandler
+import org.jetbrains.kotlin.gradle.plugin.extraProperties
+import kotlin.random.Random
 
+private val KEY_COLLECTED_DEPS = Random.nextBits(32).toString(16)
 
-fun String?.toSuffix() = this?.let { "-$it" } ?: ""
+/**
+ * Collects version information for all dependencies managed by shorthand functions (e.g. ktor, kmmresult)
+ */
+class CollectedDependencies(
+    private val project: Project
+) {
+    val versions: MutableMap<String, String> = mutableMapOf()
+    val libraries: MutableMap<String, Pair<String,String>> = mutableMapOf()
 
+    fun add(module: String, versionRef: String): String {
+        val split = module.indexOf(':')
+        val name = module.substring(split + 1)
```

Comment:
does this throw an exception when `:` is the last character of `module`?

Reply:
Good thinking. better just `split` the string and handle dependencies without explicit version numbers